### PR TITLE
#160056366 Setup redux

### PR DESCRIPTION
**Pull request by megame24**
#### What does this PR do?
Setup redux

#### Description of Task to be completed?
- install dependencies
- add components, actions, reducers, sagas folder in the src folder
- add a store.js file in the root folder
- Integrate redux-promise-middleware
- Integrate redux-saga
- Integrate redux-thunk
- write a custom error handling middleware

#### How should this be manually tested?
N/A

#### What are the relevant pivotal tracker stories?
#160056366

#### Questions:
N/A

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules
+.env
+dist/
```
Comment (chidioguejiofor) [inline .gitignore]: End OF Line

File: public/styles/App.scss [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+} 
+
+
+body{
+    color: $body-color;
+    background-color: $body-background;
+    background-image: url('../assets/beverage-books.jpg') ;
+    background-size: 100% auto;
+    background-repeat: no-repeat;
+}
```
Comment (chidioguejiofor) [inline public/styles/App.scss]: End of Line

**Review by chidioguejiofor — COMMENTED** [review]
Please correct your hound issues Once hound is merged.

Also add end of lines in the places where you missed them

File: public/styles/App.scss (hunk 2 above)
Comment (megame24) [reply public/styles/App.scss]: done

File: .gitignore (hunk 1 above)
Comment (megame24) [reply .gitignore]: done

File: src/middleware/errorMiddleware.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import isPromise from 'is-promise';
+
+const errorMiddleware = () => next => (action) => {
+  if (isPromise(action.payload)) {
+    return next(action).catch(err => err);
+  }
+  return next(action);
+};
+
+
```
Comment (tutugodfrey) [inline src/middleware/errorMiddleware.js]: please close this space here

**Review by mppdg — APPROVED** [review]

**Review by JCanaks — APPROVED** [review]

**Pull request merged**